File: check_remaining2.js
const { Client } = require('pg');
const client = new Client({ connectionString: 'postgresql://postgres:bJmkMzZQLqhzYltMNDwentYAPFUAppSq@tramway.proxy.rlwy.net:39193/railway' });

async function run() {
  await client.connect();
  
  // NounsGarhLinks: RawyID, GarhID, SayID - links narrator to grading term and say
  // narrator_criticism has narrator_id, say_id, garh_label - might cover this
  // Let's check if GarhLinks data might be embedded in narrator_criticism
  console.log('=== narrator_criticism - check if it contains GarhLinks data ===');
  console.log('Columns: id, narrator_id, scientist_noun_id, scientist_name, say_text, say_sort, say_id, garh_label');
  
  // NounsScientistsSaysLinks: SayID, ServiceMainID, ISBookTocHadith, LinkID
  // This sounds like it links critic says to book references
  // Check narrator_biography for links
  console.log('\n=== narrator_biography - quick check for book links ===');
  const nbio = await client.query(`SELECT COUNT(*) FROM narrator_biography;`);
  console.log(`narrator_biography count: ${nbio.rows[0].count}`);
  
  // NounsTranslation: NounID, ServiceMainID - links narrator to a service/book reference
  // narrator_biography has narrator_id, main_id, book_id - might map this
  const nbioSample = await client.query(`SELECT * FROM narrator_biography LIMIT 2;`);
  console.log('narrator_biography SAMPLE:', JSON.stringify(nbioSample.rows));
  
  // narrator_books
  const nbooks = await client.query(`SELECT * FROM narrator_books LIMIT 3;`);
  console.log('\nnarrator_books SAMPLE:', JSON.stringify(nbooks.rows));
  
  // Check the HadithJudgmentScientists (ID, Name) vs narrator_scientists
  // NounsScientists also exists - let's check
  console.log('\n=== NounsScientists.json exists - checking Railway narrator_scientists ===');
  const nscCols = await client.query(`SELECT column_name, data_type FROM information_schema.columns WHERE table_schema='public' AND table_name='narrator_scientists' ORDER BY ordinal_position;`);
  nscCols.rows.forEach(c => console.log(`  ${c.column_name}: ${c.data_type}`));
  
  await client.end();
}
run().catch(e => { console.error(e.message); process.exit(1); });

File: check_remaining2.py
import psycopg2, sys, io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
DB = "postgresql://postgres:bJmkMzZQLqhzYltMNDwentYAPFUAppSq@tramway.proxy.rlwy.net:39193/railway"
conn = psycopg2.connect(DB, sslmode="require")
cur = conn.cursor()

# hadith_shawahed
cur.execute("SELECT column_name, data_type FROM information_schema.columns WHERE table_name='hadith_shawahed' ORDER BY ordinal_position")
print("hadith_shawahed cols:", cur.fetchall())
cur.execute("SELECT COUNT(*) FROM hadith_shawahed")
print("hadith_shawahed count:", cur.fetchone()[0])
cur.execute("SELECT * FROM hadith_shawahed LIMIT 3")
for r in cur.fetchall():
    print("  shawahed:", r)

# hadith_index_categories
cur.execute("SELECT column_name, data_type FROM information_schema.columns WHERE table_name='hadith_index_categories' ORDER BY ordinal_position")
print("hadith_index_categories cols:", cur.fetchall())
cur.execute("SELECT COUNT(*) FROM hadith_index_categories")
print("hadith_index_categories count:", cur.fetchone()[0])
cur.execute("SELECT * FROM hadith_index_categories LIMIT 5")
for r in cur.fetchall():
    print("  idx_cat:", r)

# hadith_index_items
cur.execute("SELECT column_name, data_type FROM information_schema.columns WHERE table_name='hadith_index_items' ORDER BY ordinal_position")
print("hadith_index_items cols:", cur.fetchall())
cur.execute("SELECT COUNT(*) FROM hadith_index_items")
print("hadith_index_items count:", cur.fetchone()[0])
cur.execute("SELECT * FROM hadith_index_items LIMIT 5")
for r in cur.fetchall():
    print("  idx_item:", r)

# Check existing index pages
cur.execute("SELECT table_name FROM information_schema.tables WHERE table_name LIKE '%index%' ORDER BY table_name")
print("index tables:", [r[0] for r in cur.fetchall()])

conn.close()
